groups/bdl/bdlb/bdlb_guidutil.cpp

```diff
@@ -132,6 +137,45 @@ void GuidUtil::generate(unsigned char *result, bsl::size_t numGuids)
     }
 }
 
+void GuidUtil::generateNonSecure(unsigned char *result, bsl::size_t numGuids)
+{
+    unsigned char *bytes = result;
+    unsigned char *end   = bytes + numGuids * Guid::k_GUID_NUM_BYTES;
+
+    static bdlb::PCGRandomGenerator *pcgSingletonPtr;
+    BSLMT_ONCE_DO
+    {
+        uint64_t state;
+        if (0 != RandomDevice::getRandomBytes((unsigned char *)&state,
+                                              sizeof(state))) {
+            state = time(NULL) ^ (intptr_t)&bsl::printf;  // fallback state
+        }
```

Comment:
(for the future) This might be a candidate for factoring into local function - this is repeated twice here.

Reply:
Yes, I considered that. Also in the GuidUtil implementation, the following code is duplicated (by my new function). Perhaps a candidate for factoring as well:
    while (bytes != end) {
        typedef unsigned char uc;
        bytes[6] = uc(0x40 | (bytes[6] & 0x0F));
        bytes[8] = uc(0x80 | (bytes[8] & 0x3F));
        bytes += Guid::k_GUID_NUM_BYTES;
    }
My reasoning there was that I ought not touch working code.